tensorboard/webapp/runs/views/runs_table/regex_edit_dialog_test.ts

```diff
@@ -511,4 +511,19 @@ describe('regex_edit_dialog', () => {
     expect(list.length).toBe(6);
     expect(more.nativeElement.textContent).toBe('and 2 more');
   }));
+
+  it('makes focus stay in the regex dialog', fakeAsync(() => {
+    const fixture = createComponent(['rose']);
+    fixture.detectChanges();
+    const input = fixture.debugElement.query(By.css('input'));
+    // Works around for input element focus initial (cdkFocusInitial)
+    input.nativeElement.focus();
+
+    const div = document.createElement('div');
+    div.focus();
+    fixture.detectChanges();
+
+    expect(document.activeElement).toBe(input.nativeElement);
```

Comment:
I am surprised you don't have to wait a microtask or `tick` to assert on `activeElement`. Could you console.log test and component code to make sure this is actually working as you intended?

For example, before `div.focus`, you can do one `console.log`, put one in `handleFocusOut`, `resetFocus`, and before `expect(document.activeElement).toBe(input.nativeElement);` to see in which order things are being printed.